Assets/Scripts/Input/Player.cs

```diff
@@ -0,0 +1,217 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+/// <summary>
+/// Script controls player character behaviour
+/// </summary>
+public class Player : MonoBehaviour
+{
+    #region Fields
+
+    public float runningVelocity;
+    public float fallVelocityLimit;
+    public float jumpVelocity;
+
+    public float yAccel;
+    private float yVelo = 0f;
+    private bool grounded = false;
+    private bool jumpingGrace = false;
+    public float jumpingGraceTime;
+    private bool isLanding = false;
+    private bool cielingCollision = false;
+    public float dropFromCielingVelocity;
+    
+    private float xVelo = 0f;
+
+    public float floatMargin;
+    public float wallMargin;
+
+    //ray tracing
+    public float horizRayMargin;
+    public float downRayMargin;
+    public float upRayMargin;
+    public float bottomRayMargin;
+    public float topRayMargin;
+    private float width;
+    private float height;
+    #endregion
+
+    #region Start, Update
+
+    private void Start()
+    {
+        height = transform.localScale.y;
+        width = transform.localScale.x;
+        Gamemaster gamemaster = FindObjectOfType<Gamemaster>();
+        gamemaster.Register(this);
+    }
+
+    //Update is called once per frame
+    void Update()
+    {
+        Move(xVelo,yVelo);
+    }
+
+    #endregion
+
+    #region Physics calculation and movement resolution
+
+    /// <summary>
+    /// Interprets the movement of the player,
+    /// uses raycast to resolve movement behaviour
+    /// </summary>
+    /// <param name="xVelo">x axis velocity</param>
+    /// <param name="yVelo">y axis velocity</param>
+    private void Move(float xVelo, float yVelo)
+    {
+        //if airborne
+        if (!grounded)
+        {
+            //and rising
+            if (yVelo > 0)
+            {
+                //detect collision with above objects
+                if (CastRaysUp())
+                {
+                    cielingCollision = true;
+                }
+            }
+            //if falling, and jumpingGrace has passed
+            else if(!jumpingGrace)
+            {
+                //detect collision with the ground
+                if (CastRaysDown())
+                {
+                    Land();
+                    isLanding = true;
+                }
+            }
+        }
+
+        //if on the ground, detect falling
+        else if (!CastRaysDown())
+        {
+            grounded = false;
+        }
+
+        //resolve movement
+        transform.Translate(xVelo * Time.deltaTime, yVelo * Time.deltaTime, 0);
+
+        //post movement approximation here
+        if (isLanding)
+        {
+            ApproximateYPosition();
+            isLanding = false;
+        }
+            
+    }
+
+    private void FixedUpdate()
```

Comment:
I think all the movement/velocity calculations should be done in FixedUpdate(including the PlayerInputHandler)

Reply:
I will test this out.
Doing so might prevent the issue that the approximation functions have solved.

Reply:
Done.
Did not resolve the inconsistency: where exactly is the player detected to have collided with the wall or the floor. The approximation functions still aliviate this.